src/client/Microsoft.Identity.Client/Instance/Oidc/OidcRetrieverWithCache.cs

```diff
@@ -70,6 +75,58 @@ public static async Task<OidcMetadata> GetOidcAsync(
             }
         }
 
+        /// <summary>
+        /// Validates that the issuer in the OIDC metadata matches the authority.
+        /// </summary>
+        /// <param name="authority">The authority URL.</param>
+        /// <param name="issuer">The issuer from the OIDC metadata - the single source of truth.</param>
+        /// <exception cref="MsalServiceException">Thrown when issuer validation fails.</exception>
+        private static void ValidateIssuer(Uri authority, string issuer)
+        {
+
+            // Primary validation: check if authority begins with the issuer string (exact string comparison)
+            if (authority.AbsoluteUri.StartsWith(issuer, StringComparison.OrdinalIgnoreCase))
+            {
+                return;
+            }
+
+            // Extract tenant for CIAM-like scenarios
```

Comment:
I don't think this is needed. MSAL knows about normal CIAM authorities and will not perform OIDC discovery. I double checked this by running `ROPC_Ciam_StandardDomains_CompletesSuccessfully`